Consolidate network overview design for /home [WIP]

gozart1 opened the pull request:
* Bring the network overview design from /nexthome into /home

gozart1 commented:
Switching tacts on this a bit. Since the network overview on nexthome is much nicer looking, I'm taking a shot at consolidating that with /home.

So far, I've rebuilt the layout in /home, but without any of the css ( using only bootstrap at this point for layout & styling ). Plan is to get all the content and a fully responsive layout in place before authoring any custom styles. ( Eg. I think we can still do 3 up on mobile views instead of stacking by hiding some of the non essential stuff )

I've incorporated all the existing stats from /home except for ticket pool size. But I'm wondering if we should whittle it down a bit more. Perhaps stuff like "Tickets in Mempool" & "Votes in Mempool" can be moved to a "Mempool" section together with the "Latest Transactions" component.

This is still a WIP, but would love to get peoples thoughts.

<img width="1141" alt="screen shot 2019-01-10 at 2 55 17 am" src="https://user-images.githubusercontent.com/25571523/50964517-5b35c580-1484-11e9-94e0-594b227bf1ee.png">

Also, if we can pull this off, then the new visual blocks page can get the full page experience it deserves.

buck54321 commented:
I think this looks great. Just spitballin' a couple of random thoughts to spur discussion here. 

1. While it seems natural to put the "General" section first, it also seems like some of the least important information for a typical user of the site (whoever that guy is). I don't claim to know what is the most important information, but if I had to, I'd guess in no particular order: the ticket price, the POS and POW rewards, mining difficulty, network hashrate. 

2. I like the flat design, but I always think it looks weird when there is no distinction between the navbar/footer and main page content, i.e. white on white with no border. Its occurring in the screenshot you shared where the mining box is blending in with the footer with no clear boundary. I suppose the simple solution would be to put a light border or shadow on the navbar and footer.

chappjc commented:
"Votes in Mempool" has little to no value on the home page, but "Tickets in Mempool" is nice IMO.  Besides being useful information for helping you set your expectations for how long it might take to mine your ticket purchases, it also sort of brings the staking system to life.

gozart1 commented:
Thanks for the feedback. Agree that ticket info should be prioritized.

How changing that card's title to "Voting"? 
Seems in line with the nomenclature change from Stakepool to VSP (Voting Service Provider).

Also renamed the "General" card to "Distribution", but not sure that's the right word either. General just seems so non-descriptive to me.

I've added a subtle border to the footer and header to help delineate from main content.
It's good for performance to be sparing with css box and text shadows because they have a tendency to be expensive to render which leads to fans cranking and draining battery life.

Here's the latest from riffing on the layout a bit:

<img width="1275" alt="screen shot 2019-01-17 at 12 45 10 am" src="https://user-images.githubusercontent.com/25571523/51306327-eae1f380-19f1-11e9-9954-dc719c0f317e.png">

Mobile, night mode
<img width="399" alt="screen shot 2019-01-17 at 12 46 41 am" src="https://user-images.githubusercontent.com/25571523/51306334-ef0e1100-19f1-11e9-8d8d-68af8f950e24.png">

buck54321 commented on views/home.tmpl:
```diff
@@ -205,7 +237,7 @@
                             <a class="hash truncate-hash keyboard-target" style="flex: 1 1 auto" href="/tx/{{.Hash}}" title="{{.Hash}}">{{.Hash}}</a>
                             <span style="flex: 0 0 65px" class="mono text-right ml-1">{{.Type}}</span>
-                            <span style="flex: 0 0 105px" class="mono text-right ml-1 fs13">
+                            <span style="flex: 0 0 105px" class="mono text-right ml-1 fs13 text-secondary">
```
Your screenshot is showing a weird alignment on this "column".

gozart1 replied:
fixed, was messing around with the layout there, but ended up sticking with the inline-block approach

buck54321 commented:
The shorter progress bars look great, though the block reward progress bar seems to change size from desktop to mobile in your screenshots.  On the desktop, it looks like it would fit in a 2 x 2 grid, but in the mobile view it takes the full row width. 

After this is merged, I wonder if we could also apply a similar style to the data at the top of the address and transactions pages?

gozart1 commented:
@buck54321 I Initially gave that "Next Block Reward Reduction" progress bar some more space in the mobile via title wraps and that leads to some slight misalignment. But on second though I think letting it wrap is fine and worth it for the compactness.

Beyond that have shuffled things around a bit more
- Removed "Votes in Mempool" since that info will be displayed in mempool pane via https://github.com/decred/dcrdata/pull/961 
- Added back "Ticket Pool Size"

<img width="388" alt="screen shot 2019-01-29 at 1 47 46 am" src="https://user-images.githubusercontent.com/25571523/51899655-5750e100-2368-11e9-949c-330f2e2bb8bb.png">

<img width="1277" alt="screen shot 2019-01-29 at 1 52 06 am" src="https://user-images.githubusercontent.com/25571523/51899752-87987f80-2368-11e9-8ee3-4a4018ef494b.png">

gozart1 commented:
This is ready to merge, Here's the recap of changes:
* remove network overview controller, template & css
* rework /home to incorporate network overview design from /nexthome
* increase container max-width to 1260px
* shorten progress bar height

buck54321 commented on views/home.tmpl:
```diff
@@ -10,186 +10,222 @@
-                                    <span data-target="homepage.devFund" class="d-inline-block">{{template "decimalParts" (amountAsDecimalParts .DevFund true)}}</span
-                                    ><span class="pl-1 unit lh15rem">DCR</span>
+                            </div>
+                            <div class="col-6 col-sm-5 col-md-6 col-lg-4 mb-3 mb-sm-2 mb-md-3 mb-lg-3">
```
This one is `col-sm-5`, but all the others are `col-sm-4`. Is that intentional?

gozart1 replied:
Yes, since there's free space to the right of it at that breakpoint, giving it a bit more width prevents the label from wrapping, which in turn uses less vertical space in the layout.

buck54321 commented on views/extras.tmpl:
```diff
@@ -150,16 +150,20 @@
+        <span class="int">{{index . 0}}</span
+        >{{if gt (len (index . 1)) 0}}<span class="decimal dot">.</span
+        ><span class="decimal">{{index . 1 }}</span
+        ><span class="decimal trailing-zeroes">{{index . 2}}</span>{{end}}
```
There's a small difference in spacing here between your screenshot and what I'm seeing in chrome.

![image](https://user-images.githubusercontent.com/6109680/51906163-fbe31b00-2388-11e9-9d10-9163da90b852.png)

buck54321 commented:
This looks really good at all screen sizes. Progress bars and values seem to be updating as expected. Nice improvement all around.

gozart1 commented:
Changes made. @chappjc good spot on the annual return, I've added it back.

Not sure about encoding the time range in the linking url, maybe the default range of 1 month should be handled by that charts controller itself when no range is passed?

<img width="628" alt="screen shot 2019-01-29 at 11 34 57 am" src="https://user-images.githubusercontent.com/25571523/51935204-3fa24880-23ba-11e9-9c5e-9075d7334dda.png">
